src/labtools/adtools/finder.py

```diff
@@ -50,10 +50,23 @@ def pull_AD(read, barcoded = False, ad_preceder = "GCTAGC",
                 searched_read = re.split(bc_anteceder, roi[ad_length:], maxsplit=1)
                 if len(searched_read) == 2:
                     barcode = searched_read[0][-bclength:]
+                # LT: Increment loss_table if barcode is not found by either flanking sequence
+                else:
```

Comment:
Another way to approach this in both `pull_AD` and `pull_barcode` would be to return just the string of the key in the loss_table dictionary that should be incremented, then in `seq_counter` do something like 
`loss_table[reason_for_not_counting] += 1` 
That way you don't need to accept `loss_table` as an argument